### PR TITLE
TypeScript 2.5

### LINKED ISSUE #99: Upgrade to typescript v2.4 ?
I noticed that the current version of typescript-json-schema is pinned to 2.1.x versions of typescript. Can it be upgraded to the new v2.2.0?

### LINKED ISSUE #128: TypeScript 2.4 string enums
TypeScript 2.4 will have string enums:
https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#string-enums

Any plans to support this?

thank you

### LINKED ISSUE #146: Tons of resolution errors
Hi, 

I'm specifying a file and an interface for which to render the schema, but no matter what file I run it on, it generates a ton of errors:
image
image

Running with --ignore-errors works, but one thing I noticed is an extended interface is missing the properties of the base interface in the outputted schema, probably due to the resolution errors.

I have no issue compiling with typescript, and I set the same options in the settings: 
```
const compilerOptions: CompilerOptions = {
    strictNullChecks: true,
    "module": 1,
    "strict": true,
    "target": 4,
    "experimentalDecorators": true,
    "jsx": 2,
    "moduleResolution": 2,
}
```

TS version is 2.5.2, any help appreciated :)

### PR DESCRIPTION
**Pull request by fabiandev**
Based on the [fork](https://github.com/develar/typescript-json-schema) of @develar, I've fixed another issue with newer TypeScript versions and all tests are passing. The TypeScript version is now at ~2.5.3.   This closes #99, closes #128 and resolves #146.

### CODE REVIEW

**Comment by domoritz** [comment]
Oh sweet. Can you rebase to master so we can merge without conflicts?

**Comment by fabiandev** [comment]
Thanks! Does the [rebase and merge](https://github.com/blog/2243-rebase-and-merge-pull-requests) option work for you?

**Comment by domoritz** [comment]
screenshot_20171005-094511

File: typescript-json-schema.ts [hunk 1]
```diff
@@ -344,9 +345,9 @@ export class JsonSchemaGenerator {
         const reffedType = this.getReferencedTypeSymbol(prop, tc);
 
         let definition = this.getTypeDefinition(propertyType, tc, undefined, undefined, prop, reffedType);
-        if (this.args.titles) {
-            definition.title = propertyName;
-        }
+        // if (this.args.titles) {
```
Comment (domoritz) [inline typescript-json-schema.ts]: What's up with this?

File: typescript-json-schema.ts [hunk 2]
```diff
@@ -63,14 +63,14 @@ export type Definition = {
     type?: string | string[],
     definitions?: {[key: string]: any},
     format?: string,
-    items?: Definition,
+    items?: Definition | Array<Definition>,
```
Comment (domoritz) [inline typescript-json-schema.ts]: Let's make this consistent and use `[]` instead of `Array`. Can you also correct the defs below?

**Comment by fabiandev** [comment]
Is the PR okay now, or is there anything left to do? Looks like there should be no merge conflicts.

**Comment by domoritz** [comment]
Done

**Pull request merged**